### PR TITLE
Restore for external PG 

### PR DESCRIPTION
**Pull request by jaym**
Make restore work with external PG. This introduces a new database backup operation and deprecates the old one, meaning for external PG backup/restore to work, you'll need to use the latest automate. The reason we need a new op is because we now use the custom pg dump format. This allows us to ignore restoring permissions, which is important because we no longer get to always control the role to use. These will get fixed up when services start up and ask pg-helper to do things.

### CODE REVIEW

File: lib/platform/pg/exporter.go [hunk 1]
```diff
@@ -18,20 +18,25 @@ import (
...
+	DataDir           string
+	Name              string
+	User              string
+	IncludedTables    []string
+	ExcludedTables    []string
+	ConnInfo          ConnInfo
+	CmdExecutor       command.Executor
+	Timeout           time.Duration
+	DisableRoleCreate bool
+	UseCustomFormat   bool
```
[the remark this answers is not in the document]
Comment (jaym) [reply lib/platform/pg/exporter.go]: we can do that but we must still keep it around for "compatibility" with old backups

File: lib/platform/pg/exporter_test.go [hunk 2]
```diff
@@ -184,9 +188,10 @@ func TestImport(t *testing.T) {
 		ioutil.WriteFile(exportFile, testExportContent, 0700)
 
 		mockExec.Expect("Run", command.ExpectedCommand{
-			Cmd:  "hab",
-			Args: []string{"pkg", "exec", "chef/automate-postgresql", "psql", "-f", exportFile, "-v", "ON_ERROR_STOP=true"},
-			Env:  append(testExpectedEnv, "PGDATABASE=test_database"),
+			Cmd: "hab",
+			Args: []string{"pkg", "exec", "chef/automate-postgresql", "psql", "-f", exportFile,
+				"-v", "ON_ERROR_STOP=true", connURI("test_database")},
+			Env: append(testExpectedEnv, "PGDATABASE=test_database"),
```
[the remark this answers is not in the document]
Comment (jaym) [reply lib/platform/pg/exporter_test.go]: yea, wtf

File: lib/platform/pg/exporter.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (stevendanna) [reply lib/platform/pg/exporter.go]: 🤦‍♂️ Yes, of course

File: lib/platform/pg/exporter_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaym) [reply lib/platform/pg/exporter_test.go]: it seems to not care about extra values in the array

[the remark this answers is not in the document]
Comment (jaym) [reply lib/platform/pg/exporter_test.go]: i should stop passing the environment. it's all included in the connection uri string

[the remark this answers is not in the document]
Comment (jaym) [reply lib/platform/pg/exporter_test.go]: cant remove it because i think the a1 conninfo stuff will want to pass PGPASSFILE

**Comment by jaym** [comment]
I undid the a1 migration changes. It requires rebuilding our test packages

**Pull request merged**